pg_search/src/postgres/storage/metadata.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2023-2025 ParadeDB, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use crate::postgres::storage::block::{BM25PageSpecialData, SegmentMetaEntry};
+use crate::postgres::storage::block::{LinkedList, METADATA};
+use crate::postgres::storage::buffer::{Buffer, BufferManager, BufferMut};
+use crate::postgres::storage::merge::{
+    MergeEntry, MergeList, MergeLock, SegmentIdBytes, VacuumList, VacuumSentinel,
+};
+use crate::postgres::storage::{LinkedBytesList, LinkedItemList};
+use pgrx::pg_sys;
+use tantivy::index::SegmentId;
+
+/// The metadata stored on the [`Metadata`] page
+#[derive(Debug, Copy, Clone)]
+#[repr(C, packed)]
+pub struct MetaPageData {
+    /// This space was once used but no longer is.  As such, it needs to remain dead forever
+    #[allow(dead_code)]
+    pub _dead_space: [u32; 2],
+
+    /// Contains the [`pg_sys::BlockNumber`] of the active merge list
+    active_vacuum_list: pg_sys::BlockNumber,
+
+    /// A block for which is pin is held during `ambulkdelete()`
+    pub ambulkdelete_sentinel: pg_sys::BlockNumber,
+
+    /// The header block for a [`LinkedItemsList<MergeEntry>]`
+    pub merge_list: pg_sys::BlockNumber,
```

Comment:
As I said above, this one should stay inside MergeLock.  It's not at all safe/correct to access the merge list in any way without holding the MergeLock and the code should enforce that.